src/Extensions/Abstractions/Validation.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Omex.Extensions.Abstractions
+{
+	/// <summary>
+	/// Class provides common validation methods
+	/// </summary>
+	public static class Validation
+	{
+		/// <summary>
+		/// Throw exception if value is null, otherwise returns not-nullable value
+		/// </summary>
+		/// <typeparam name="T">type of the value</typeparam>
+		/// <param name="value">value to check</param>
+		/// <param name="name">name of the value for exception message</param>
+		/// <exception cref="ArgumentNullException">if value is null</exception>
+		[MethodImpl(MethodImplOptions.AggressiveInlining)]
+		public static T ThrowIfNull<T>(T? value, string? name = null) where T : class
+			=> value ?? throw new ArgumentNullException(name);
```

Comment:
what happens here if `name` is null`?

Reply:
If the name is `null` message would be "Value cannot be null". https://sharplab.io/#v2:C4LgTgrgdgPgAgJgIwFgBQiAM7EGYAEA3uvqfogkSWTcABZgD2A7vlAKasCCYA5hAFt2UYADkIAGwkBRAB4BjdgAdgAS0ZQAFFEkSAlAG5qpAL7oTQA=
If the name is "test" message would be "Value cannot be null. (Parameter 'test')" https://sharplab.io/#v2:C4LgTgrgdgPgAgJgIwFgBQiAM7EGYAEA3uvqfogkSWTcABZgD2A7vlAKasCCYA5hAFt2UYADkIAGwkBRAB4BjdgAdgAS0ZQAFACJg7AM7BtASgDc1UgF90loA===